types/ziggy-js/index.d.ts

```diff
@@ -0,0 +1,55 @@
+// Type definitions for ziggy-js 0.9
+// Project: https://github.com/tightenco/ziggy#readme
+// Definitions by: Ben Allfree <https://github.com/benallfree>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.6
+
+export type EncodeURIParamType = string | number | boolean;
+
+export type Params =
+    | {
+          [key: string]: EncodeURIParamType;
+      }
+    | EncodeURIParamType
+    | EncodeURIParamType[];
+
+export interface Route {
+    uri: string;
+    methods: Array<'GET' | 'HEAD' | 'POST' | 'PATCH' | 'PUT' | 'OPTIONS' | 'DELETE'>;
+    domain?: null | string;
+    middleware?: any[];
+    bindings?: any[];
+}
+
+export interface Config {
+    namedRoutes: {
+        [key: string]: Route;
+    };
+    baseUrl: string;
+    baseProtocol: 'http' | 'https';
+    baseDomain: string;
+    basePort?: number | null;
+    defaultParameters: {
+        [_: string]: string;
+    };
```

Comment:
@bakerkretzmar Just a side comment, in `route.js` I think it might be nice to do 

```
                if (this.ziggy.defaultParameters.hasOwnProperty(keyName)) {
                    defaultParameter = encodeURIComponent(this.ziggy.defaultParameters[keyName]);
                }
```

That way default params could be any encodable type.

Reply:
Default parameters actually can only be strings and integers, they aren't subject to route model binding: see https://laravel.com/docs/8.x/urls#default-values at the "URL Defaults & Middleware Priority" heading, and https://github.com/laravel/framework/blob/8.x/src/Illuminate/Routing/RouteUrlGenerator.php#L211-L231

Reply:
Ok I've updated it to `string | number`. Is the goal to mimic the PHP `route()` as much as possible? TS doesn't have an `int` type, so the best we can do is `number`. Hope that works :)

Reply:
Yep that's the goal, and `number` is great thanks!